### DISCUSSION TITLE
Character Name exists. Making new Name

### DISCUSSION THREAD

**Discussion by chule74** [isid0re/SoloLeveling#153] [Q&A]
Hi. I'm a Korean. I play on the Asian server, but when the bot level up to some extent, it creates a new character. Can't you fix this?
In Korea, Asian servers are down several times a day.
After the realm down, there are several attempts to connect, the delay proceeds, and the character is regenerated when the realm down is released.

**Comment by isid0re** [comment]
There are a lot of variables at play with what you are describing. I would need to actually see it happen to see how I can go about that. If you can provide a recording of what happens I'd be willing to see about fixing it.

Also be sure you are running the latest version. I previously fixed an issue regarding remaking characters when the login couldn't find the character because it had to scroll down to find the character

**Reply by chule74** [reply]
Thank you for your reply.
I think..
If the bot is restarted due to realmdown, the connection attempt will continue. It seems that the profile json file in the data folder cannot be read when the connection is made after realmdown is released. So it seems to be making a new character.
When creating a new character, Character Name exists. The message Making new Name appears. console

It is the same phenomenon for all Koreans.

**Reply by isid0re** [reply]
Can you please download and update the files with the available version after this post and let me know if this still occurs. Thanks.

**Reply by chule74** [reply]
I downloaded and ran it again yesterday and the same thing happened. When is the update?

**Reply by chule74** [reply]
In my opinion, when restarting after realmdown, when loading the data folder profile.json file, the character information is not receiving properly. I'm not sure if it's related because after downloading Realm, if I keep trying to log in, I get a message related to a login error.

**Reply by isid0re** [reply]
Ok if you've downloaded from 1 to 2 days ago ( difference between Korea and me) then the update for more than 8 characters should already be there. 

So, are you using a randomly generated account ? Or are you using a personal account you already have?

This can help us narrow down where the problem is.

**Reply by chule74** [reply]
Both random and existing accounts are the same. Asian servers in Korea regularly meet Realmdown 10 times a day. Because of that, the server is unstable.

**Reply by k8k901007** [reply]
안녕하세요 chule74님.
isd0re님한테 연락 받고 댓글 남깁니다.
번역하시는데에 있어서 어려움을 격으시는거 같더라고요.
아시아 렐름 외에 현재 다른 렐름에는 렐름다운 증상이 없어서 무엇을 말씀하시는지 잘 모르실겁니다.

해결 방법은 2가지 있습니다.

1. D2botLead.dbj를 여신후 상단 부분에 RealmDownDelay: 3, 아래에 InvalidPasswordDelay: 15, 를 추가해주시고
Ctrl+F로 case getLocaleString(5207): 해당 구문을 검색하시면

			case getLocaleString(5207):
				D2Bot.updateStatus("Invalid Password");
				D2Bot.printToConsole("Invalid Password");
				//createAccount = true;

				break;
이런식으로 되어있는 구문을

			case getLocaleString(5207):
				D2Bot.updateStatus("Invalid Password");
				D2Bot.printToConsole("Invalid Password");
				
				ControlAction.timeoutDelay("Invalid Password", StarterConfig.InvalidPasswordDelay * 6e4);
				
				D2Bot.printToConsole("Invalid Password - Restart");
				D2Bot.restart();
				
				break;
이렇게 변경하는 방법.

2. d2bs 매니저창 보시면 Schedule란에 
02:15 ~ 02:45 
04:00 ~ 04:25
05:15 ~ 05:45 
08:15 ~ 08:45 
11:15 ~ 11:45 
14:15 ~ 14:45 
17:15 ~ 17:45 
20:15 ~ 20:45 
23:15 ~ 23:45
해당 시간대 마다 봇을 자동 재부팅 시키는 방법 2가지입니다.

보통 첫번째 방법을 많이 사용하십니다.
진입점(Script Entry)이 다른거면 해당 진입점을 위와 동일한 방법으로 수정 해주시면 됩니다.
참고 되셨으면 합니다.

**Reply by chule74** [reply]
위에방법은 이미 사용하고 있는겁니다. 그방법과는 별도의 문제같아요. 한국에서 아시아서버를 이용할시에 하루에 몇번이나 정기적으로 렐름다운이 있습니다. 서버가불안정한건지 모르겠지만 그냥 거희 4시간마다? 렐름다운이 있다고 보시면되요. 렐름다운은 20분정도 유지되고 그이후에 렐름다운이 해제됩니다. 그럼 봇이 자동으로 계정에 재접속시도를 합니다. 그래서 위에 남겨주신함수처럼 패스워드 에러가나면 대기열시간을 줘서 대기를하고 리스타트를 하게되겠죠 그건이미 사용하고있습니다. 하지만 재접속을 하게되면 캐릭터를 새로만들어요 data폴더안에 있는 계정정보를 받아오질못하는거같아요.

**Reply by k8k901007** [reply]
https://discord.gg/2JAAWBCQnb
카카오톡 한개 주시겠어요?

**Reply by isid0re** [reply]
Thanks LuC

**Reply by deux2303** [reply]
I am also Korean. I also have the same problem. There are a lot of Lelm Downs in Korea. In Korea, it happens 10 to 12 times a day. The time is random, with 10 to 40 minutes. In the meantime, that try to access characters. However, since they couldn't access it, they create a new character and restart it when that done with it.

**Comment by isid0re** [comment]
the discord server has translation from korean to english and english to korean. if you have any further questions or needs please join me there.  https://discord.gg/2JAAWBCQnb
